graphql_compiler/compiler/ir_lowering_match/utils.py

```diff
@@ -222,12 +222,17 @@ def construct_where_filter_predicate(simple_optional_root_info):
 CompoundMatchQuery = namedtuple('CompoundMatchQuery', ('match_queries'))
 
 
-class OptionalTraversalTrie(dict):
-    def __init__(self):
+class OptionalTraversalTrie():
```

Comment:
I think this class needs much more documentation -- I only understand what it does because we discussed it in front of a whiteboard. I also wouldn't count on people knowing what `complex_optional_roots` is.